kombu/transport/redis.py

```diff
@@ -829,23 +831,45 @@ def get_table(self, exchange):
                 raise InconsistencyError(NO_ROUTE_ERROR.format(exchange, key))
             return [tuple(bytes_to_str(val).split(self.sep)) for val in values]
 
-    def _lookup_direct(self, exchange, routing_key):
-        if not exchange:
-            return [routing_key]
+    def _lookup(self, exchange, routing_key, default=None):
+        """Find all queues matching `routing_key` for the given `exchange`.
 
-        key = self.keyprefix_queue % exchange
-        pattern = ''
-        queue = routing_key
-        queue_bind = self.sep.join([
-            routing_key or '',
-            pattern,
-            queue or '',
-        ])
-        with self.conn_or_acquire() as client:
-            if client.sismember(key, queue_bind):
-                return [queue]
+        Returns:
+            str: queue name -- must return the string `default`
+                if no queues matched.
+        """
+        result = []
+        if default is None:
+            default = self.deadletter_queue
+        if not exchange:  # anon exchange
+            return [routing_key or default]
+
+        try:
+            pattern = ''
+            queue = routing_key
+            redis_key = self.keyprefix_queue % exchange
+            queue_bind = self.sep.join([
+                routing_key or '',
+                pattern,
+                queue or '',
+            ])
+            with self.conn_or_acquire() as client:
+                if client.sismember(redis_key, queue_bind):
+                    result = [queue]
+                if not client.scard(redis_key):
```

Comment:
Why is this below `sismember`?
Won't `sismember` return false anyway if `redis_key`'s cardinality is zero?

Reply:
Yup good catch -- I mean't for it to go first.